### PR TITLE
Upgrade version of react-on-rails to 3

### PR DESCRIPTION
**Pull request by jbhatab**
[Review on Reviewable](https://reviewable.io/reviews/shakacode/react-webpack-rails-tutorial/224)

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -37,7 +37,7 @@ gem "sdoc", group: :doc
 # Use Rails Html Sanitizer for HTML sanitization
 gem "rails-html-sanitizer"
 
-gem "react_on_rails"
+gem "react_on_rails", git: "https://github.com/shakacode/react_on_rails.git", tag: "3.0.0-beta.1"
```
Comment (justin808) [inline Gemfile]: why do this when the gem is on rubygems?

**Comment by justin808** [comment]
Need to get CI to pass, and use rubygems version of beta.

---

Reviewed 4 of 4 files at r1, 13 of 13 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/shakacode/react-webpack-rails-tutorial/224#-:-KA4BQoCJtvq5vj3BPh5:871269859)*

**Comment by justin808** [comment]
@jbhatab Any updates on this one?

File: Gemfile (hunk 1 above)
Comment (jbhatab) [inline Gemfile]: @justin808 You need git defined to use tag.

**Comment by justin808** [comment]
@alexfedoseev If you can, take a peek at this one. Odd that we get the error here with react-router.

**Comment by justin808** [comment]
@jbhatab I investigated with Alex. 

We should update to react-router v2 final and double check our API usage. The problem is probably due to the beta version that in the tutorial.



to update the deps, 

```
rm npm-shrinkwrap.json
ncu -a -u
npm i
```

File: client/app/bundles/comments/store/routerCommentsStore.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  const initialState = {
+    $$commentsStore: $$commentsState.merge({
+      $$comments: initialComments,
+    }),
+  };
+
+  // https://github.com/reactjs/react-router-redux
+  const reducer = combineReducers({
+    ...reducers,
+    routing: routerReducer,
```
Comment (justin808) [inline client/app/bundles/comments/store/routerCommentsStore.js]: @alexfedoseev does it make sense to have a different store for the client with router, as compared to the server store? Maybe it doesn't matter?

Possibly the routeReducer would expect the history.

File: client/app/libs/middlewares/loggerMiddleware.js [hunk 3]
```diff
@@ -8,11 +8,13 @@ export default function logger({ getState }) {
...
-    const immutableState = getState();
+    const state = getState();
     const readableState = {};
-    for (const storeItem in immutableState) {
-      if (immutableState.hasOwnProperty(storeItem)) {
-        readableState[storeItem] = immutableState[storeItem].toJS();
+    for (const storeItem in state) {
+      if (state.hasOwnProperty(storeItem)) {
+        readableState[storeItem] = (
+          state[storeItem].toJS ? state[storeItem].toJS() : state[storeItem]
```
Comment (justin808) [inline client/app/libs/middlewares/loggerMiddleware.js]: @alexfedoseev This is b/c the react-router-redux state is a plain object.

File: client/app/bundles/comments/store/routerCommentsStore.js (hunk 2 above)
Comment (alexfedoseev) [inline client/app/bundles/comments/store/routerCommentsStore.js]: I don't think we need different store. This is `routeReducer`:

```js
// Reducer

const initialState = {
  location: undefined
}

export function routeReducer(state = initialState, { type, payload: location }) {
  if (type !== UPDATE_LOCATION) {
    return state
  }

  return { ...state, location }
}
```

File: client/app/libs/middlewares/loggerMiddleware.js (hunk 3 above)
Comment (alexfedoseev) [inline client/app/libs/middlewares/loggerMiddleware.js]: :+1:

File: client/app/bundles/comments/store/routerCommentsStore.js (hunk 2 above)
Comment (justin808) [inline client/app/bundles/comments/store/routerCommentsStore.js]: I removed that file (local one) unless you're talking about the one from react-router-redux.

Probably doesn't hurt, right, to have both?

Comment (justin808) [inline client/app/bundles/comments/store/routerCommentsStore.js]: But maybe better to remove.

Comment (justin808) [inline client/app/bundles/comments/store/routerCommentsStore.js]: Since this PR is already merged, feel free to remove on the next PR, @alexfedoseev

**Pull request merged**